isis/tests/TgoCassisModuleTests.cpp

```diff
@@ -2178,3 +2178,593 @@ TEST(TgoCassisModuleTests, TgoCassisUncontrolledSingleColorMosaic) {
+  UserInterface tgocassis2isisNir(TGOCASSIS2ISIS_XML, tgocassis2isisArgs);
+  try {
+    tgocassis2isis(tgocassis2isisNir);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to run tgocassis2isis on nir image: " << e.what() << std::endl;
+  }
+
+  spiceinitArgs = {"from=" + nirFileName,
+                   "ck=" + binaryCkKernelsAsString,
+                   "spk=" + binarySpkKernelsAsString};
+  UserInterface spiceinitNir(SPICEINIT_XML, spiceinitArgs);
+  try {
+    spiceinit(spiceinitNir);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to run spiceinit  on nir image: " << e.what() << std::endl;
+  }
+
+  // run stitch and unstitch on cube list
+  FileList *cubeList = new FileList();
+  cubeList->append(panFileName);
+  cubeList->append(bluFileName);
+  cubeList->append(redFileName);
+  cubeList->append(nirFileName);
+
+  QString cubeListFile = prefix.path() + "/cubelist.lis";
+  cubeList->write(cubeListFile);
+
+  QVector<QString> stitchArgs = {"fromlist=" + cubeListFile,
+                "outputprefix=" + prefix.path() + "/stitched"};
+  UserInterface stitchOptions(STITCH_XML, stitchArgs);
+
+  try {
+    tgocassisstitch(stitchOptions);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to run tgocassisstitch with cube list: " << e.what() << std::endl;
+  }
+
+  // run mosrange on cube list
+  QString tgocassisstitchOutput = prefix.path() + "/stitched-2016-11-26T22:50:27.381.cub";
+  FileList *mosrangeCubeList = new FileList();
+  mosrangeCubeList->append(tgocassisstitchOutput);
+
+  QString mosrangeCubeListFile = prefix.path() + "/cubelist.lis";
+  mosrangeCubeList->write(mosrangeCubeListFile);
+
+  QString mapFile = prefix.path() + "/stitched.map";
+  QVector<QString> mosrangeArgs = {"fromlist=" + mosrangeCubeListFile, "to=" + mapFile};
+  UserInterface mosrangeOptions(MOSRANGE_XML, mosrangeArgs);
+
+  try {
+    mosrange(mosrangeOptions);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to run mosrange with cube list: " << e.what() << std::endl;
+  }
+
+  // run cam2map and cassismos on pan cube
+  QString projectedFile = prefix.path() + "/projected.cub";
+  QVector<QString> cam2mapArgs = {"from=" + tgocassisstitchOutput,
+                                  "to=" + projectedFile,
+                                  "map=" + mapFile,};
+  UserInterface cam2mapPan(CAM2MAP_XML, cam2mapArgs);
+  try {
+    cam2map(cam2mapPan);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to run cam2map on stitched image: " << e.what() << std::endl;
+  }
+
+  // run tgocassisrdrgen on red framelet.
+  QVector<QString> rdrgenArgs = {"from=" + projectedFile,
+                                 "to=" + prefix.path() + "/exported.img"};
+  UserInterface rdrgen(RDRGEN_XML, rdrgenArgs);
+  try {
+    tgocassisrdrgen(rdrgen);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to run tgocassisrdrgen on projected image: " << e.what() << std::endl;
+  }
+
+  // QString ingestedExportFile = prefix.path() + "/exported.cub";
```

Comment:
Can we remove the commented code or is there a reason to keep it around?

Reply:
We need to be able to check that the output from tgocassisrdrgen is valid. This is commented out as we originally thought the output product could just be ingested but this turn out to be incorrect.

I will put in an issue to get this tested properly. Setting up the infrastructure to test this properly is outside the scope of support hours